src/components/tree-detail/tree-adopt-card.tsx

```diff
@@ -11,43 +13,119 @@ const TreeAdoptCard: React.FC<TreeAdoptCardProps> = ({
   treeData,
   treeAgeClassification,
 }) => {
+  const [showTooltip, setShowTooltip] = useState(false);
+  const [heartHovered, setHeartHovered] = useState(false);
   const i18n = useI18nStore().i18n();
+
+  const { adoptTree, unadoptTree, isAdopted, isLoading, adoptedByOthers } =
+    useAdoptTree(treeData.id);
+
   return (
     <div className="shadow-gdk-hard flex flex-col gap-4 rounded-lg bg-slate-100 p-4">
       <div className="flex flex-row items-center justify-between text-xl">
         <div className="font-bold">{treeData.artdtsch}</div>
-        <img
-          src="/images/heart-icon.svg"
-          alt="Tree Icon"
-          width={24}
-          height={24}
-        />
-      </div>
-      {treeAgeClassification === TreeAgeClassification.BABY && (
-        <div>{i18n.treeDetail.managedBy}</div>
-      )}
-      {treeAgeClassification !== TreeAgeClassification.BABY && (
-        <div className="flex flex-col gap-4">
-          <div className="flex flex-row items-center justify-between">
-            <div className="text-slate-500">{i18n.treeDetail.adoptIt}</div>
+
+        <button
+          onClick={async () => {
+            if (!isAdopted) {
+              await adoptTree();
```

Comment:
These functions throw errors and these errors are not handled

Reply:
handled with global error handler, see last commit